core/src/main/java/org/elasticsearch/search/aggregations/metrics/cardinality/HyperLogLogPlusPlus.java

```diff
@@ -162,8 +159,12 @@ public static long memoryUsage(int precision) {
     private final double alphaMM;
 
     public HyperLogLogPlusPlus(int precision, BigArrays bigArrays, long initialBucketCount) {
-        Preconditions.checkArgument(precision >= 4, "precision must be >= 4");
-        Preconditions.checkArgument(precision <= 18, "precision must be <= 18");
+        if (precision < 4) {
```

Comment:
I might take the opportunity to roll these into one statement like:
```java
if (precision < 4 || 18 < precision) {
  throw new IllegalArgumentException("precision must be between 4 and 18 inclusive");
}
```
I figure it gets more information back to the user on failure. Not a big deal though.